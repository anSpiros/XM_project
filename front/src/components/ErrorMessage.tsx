import React from "react";
import { ErrorMessageInterface } from "../interfaces/error-message.interface";

function ErrorMessage({ message }: ErrorMessageInterface) {
  return (
    <>
      {message && (
        <p data-testid="error-message" className="error-message text-danger">
          {message}
        </p>
      )}
    </>
  );
}

export default ErrorMessage;
