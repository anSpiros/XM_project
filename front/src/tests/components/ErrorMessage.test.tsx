import React from "react";
import { render, screen } from "@testing-library/react";
import ErrorMessage from "../../components/ErrorMessage";

describe("Error message component", () => {
  test("should render correctly", () => {
    const { asFragment } = render(<ErrorMessage message={"test"} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render empty", () => {
    const { asFragment } = render(<ErrorMessage message={""} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("text should appear on screen", () => {
    render(<ErrorMessage message={"test error message"} />);
    expect(screen.getByText(/test error message/i)).toBeInTheDocument();
  });
});
