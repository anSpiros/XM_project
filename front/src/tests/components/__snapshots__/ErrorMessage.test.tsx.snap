// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error message component should render correctly 1`] = `
<DocumentFragment>
  <p
    class="error-message text-danger"
    data-testid="error-message"
  >
    test
  </p>
</DocumentFragment>
`;

exports[`Error message component should render empty 1`] = `<DocumentFragment />`;
